autonat.go

```diff
@@ -190,6 +193,13 @@ func (as *AmbientAutoNAT) background() {
 				}
 			}
 
+		// peer identification occured.
+		case idev := <-IDChan:
+			peer = idev.(event.EvtPeerIdentificationCompleted).Peer
+			if s, err := as.host.Peerstore().SupportsProtocols(peer, AutoNATProto); len(s) == 0 || err != nil {
```

Comment:
@willscott 

We will receive from both the  `inboundConn ` AND  `IDChan ` channels for the same inbound connection here and could end up calling `scheduleProbe` twice for the same connection.

Would this be detrimental or can we live with it ?